### ISSUE TITLE
Simplify string types to just string

### ISSUE THREAD

**Issue by mitar** [deepmap/oapi-codegen#277]
I do not know why generated types are so precise, but I think at least for string types, they could parameters and response values could simply be `string`, and not custom types like `type ClientIdParameter string`. I do not get what is gained by the fact that I have to then call `client.GetClientWithResponse(ctx, ClientIdParameter(clientId))`? And then similarly, the response is also a new type. So I have to convert it back to regular string, or even if I want to pass it to another API call, convert it to another string-like type. A lot of typing and I am not sure why?

**Comment by deepmap-marcinr** [comment]
Agreed. I'd like to do this in the future.

**Comment by mitar** [comment]
One simple way to fix this would be that instead of:

```
type ClientIdParameter string
```

code would be:

```
type ClientIdParameter = string
```

Then it is still similarly readable when you read generated code and you can easier find the corresponding definition in the spec, but from Go perspective, this is then exactly the same as `string`.

**Comment by ForsakenHarmony** [comment]
The main benefit of this is enums I suppose, but not sure if the case you ran into was an enum

**Comment by mitar** [comment]
No, for me it is about strings themselves. For enums is maybe even reasonable to have a special type, I would say. But for raw strings I do not see much benefit.

**Comment by mromaszewicz** [comment]
This has been addressed. All primitive types (string, int, bool, float, etc.) now set DefineViaAlias = true in schema.go, which generates type aliases like:

  type ClientIdParameter = string

instead of type definitions like:

  type ClientIdParameter string

Type aliases are fully interchangeable with the underlying type in Go, so no conversion is needed when passing values between API calls.

Note: the old behavior can be restored via the old-aliasing compatibility option.

Closing as completed.
